src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1004,17 +1008,19 @@ protected virtual void MarkUserDependency (IMemberDefinition context, CustomAttr
 				return;
 			}
 
-			if (MarkDependencyMethod (td, member, signature, new DependencyInfo (DependencyKind.PreservedDependency, ca)))
-				return;
+			if (member != null) {
+				if (MarkDependencyMethod (td, member, signature, new DependencyInfo (DependencyKind.PreservedDependency, ca)))
+					return;
 
-			if (MarkNamedField (td, member, new DependencyInfo (DependencyKind.PreservedDependency, ca)))
-				return;
+				if (MarkNamedField (td, member, new DependencyInfo (DependencyKind.PreservedDependency, ca)))
+					return;
+			}
 
-			_context.LogWarning (
+			Context.LogWarning (
 				$"Could not resolve dependency member '{member}' declared in type '{td.GetDisplayName ()}' specified in a 'PreserveDependency' attribute.", 2005, context);
```

Comment:
We will print empty here in the case member is null, not sure if it's important or if we need a specific error code for that

Reply:
I think it's ok since this is the same thing that would be printed if the member signature were an empty string, and because PreserveDependency is deprecated so this entire code path is legacy anyway.